### ISSUE TITLE
How to populate abstracts?

### ISSUE THREAD

**Issue by DOFfactory** [JabRef/jabref#8208]
### JabRef version

5.3 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 10

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

I would like to mass-populate the abstracts of bibliography entries

1. Select entries
2. Right click > Get blibliographic data from DOI/ISBN/...
3. Data populates, but no abstract
3. Right click on the same entry > Open URL or DOI
4. Webpage loads - there is an abstract available

### Appendix

N/A

**Comment by koppor** [comment]
We use external services. For DOI fetching, JabRef uses the crossref service (see also https://docs.jabref.org/faq#q-why-cant-jabref-find-any-doi-isbn-eprint-etc.)

One could try to implement fetching abstract functionality to the existing fetchers (https://devdocs.jabref.org/advanced-reading/fetchers) and then craft a composite fetcher first fetching the DOI and then fetching the abstract from the publisher where the paper belongs so.

**Comment by Siedlerchr** [comment]
The JabRef browser extension is helpful for importing data, if often includes the abstract as well

**Comment by DOFfactory** [comment]
Thanks @Siedlerchr, this is certainly possibly, but what I had in mind is more the "Add new entry" option, when one searches for a single DOI, and then imports the additional entry infos. 

![image](https://user-images.githubusercontent.com/10846169/151699013-b63f355c-7978-4031-af2a-5e170b064306.png)

This should also include the abstract, because currently it does not. One has to click on the links after import, and copy&paste the abstract manually (including all the other missing data, e.g. keywords).

**Comment by Siedlerchr** [comment]
Yes, I know this is unfortunate, but the doi.org response gives us only the BibTeX back without the abstract. I can only guess, but it may have been something to do with the fact that abstracts are copyrighted.

**Comment by alfureu** [comment]
just bumped into this, can we maybe use services like semanticscholar or similar to get the abstracts? it is really a pain not having them

**Comment by koppor** [comment]
When implementing, I think, a new AbstractFetcher needs to be created.

Hints on implementation at https://devdocs.jabref.org/code-howtos/fetchers.html

Note that we have an EntryBasedFetcher to complete an entry. However, I don't know how this is accessible in the UI.

Finally, some fetchers fetch the abstracts. We only remove them after fetching in test cases.

**Comment by ryan-carpenter** [comment]
References without abstracts are still a problem. Copyright is only an issue for abstracts from specific databases. JabRef should be able to add a publicly available abstract to an existing entry, even if the abstract comes from a different source than the rest of the entry. The most common scenario for me would be references added to my library via DOI (without an abstract) that also exist in PubMed (with an abstract). JabRef could retrieve the abstracts from PubMed.

You could consider this issue more broadly, because abstracts are not the only additional information that a second source might have to add. JabRef's feature could be something like "Supplement with ... " Or "Enhance with ... " choose a source. Personally, I like having the accurate publishing information from the publisher merged with the abstract, indexing, PMID, and PMCID from PubMed.
